Content.Server/_Kaif/Xenomorphs/Systems/AlienQueenSystem.cs

```diff
@@ -0,0 +1,116 @@
+﻿using Content.Server.Actions;
+using Content.Server.Aliens.Components;
+using Content.Server.Animals.Components;
+using Content.Server.Popups;
+using Content.Shared.Aliens.Components;
+using Content.Shared.Aliens.Systems;
+using Content.Shared.Coordinates.Helpers;
+using Content.Shared.Hands.Components;
+using Content.Shared.Maps;
+using Content.Shared.Physics;
+using Content.Shared.Tag;
+using Robust.Shared.Containers;
+using Robust.Shared.Map;
+using AlienEggActionEvent = Content.Shared.Aliens.Components.AlienEggActionEvent;
+using AlienQueenComponent = Content.Shared.Aliens.Components.AlienQueenComponent;
+
+namespace Content.Server.Aliens.Systems;
+
+/// <summary>
+/// This handles...
+/// </summary>
+public sealed class AlienQueenSystem : EntitySystem
+{
+    [Dependency] private readonly AlienEvolutionSystem _evolution = default!;
+    [Dependency] private readonly PopupSystem _popup = default!;
+    [Dependency] private readonly ActionsSystem _actions = default!;
+    [Dependency] private readonly EntityLookupSystem _lookupSystem = default!;
+    [Dependency] private readonly TurfSystem _turf = default!;
+    [Dependency] private readonly IMapManager _mapMan = default!;
+    [Dependency] private readonly SharedContainerSystem _container = default!;
+    [Dependency] private readonly SharedPlasmaVesselSystem _plasmaVessel = default!;
+    /// <inheritdoc/>
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<AlienQueenComponent, MapInitEvent>(OnMapInit);
+        SubscribeLocalEvent<AlienQueenComponent, AlienEggActionEvent>(OnEgg);
+        SubscribeLocalEvent<AlienQueenComponent, RoyalLarvaActionEvent>(OnRoyalLarva);
+    }
+
+    private void OnMapInit(EntityUid uid, AlienQueenComponent component, MapInitEvent args)
+    {
+        _actions.AddAction(uid, ref component.EggActionEntity, component.EggAction);
+    }
+
+    private void OnEgg(EntityUid uid, AlienQueenComponent component, AlienEggActionEvent args)
+    {
+        if (TryComp<PlasmaVesselComponent>(uid, out var plasmaComp)
+            && plasmaComp.Plasma < component.PlasmaCostEgg)
+        {
+            _popup.PopupClient(Loc.GetString(Loc.GetString("alien-action-fail-plasma")), uid, uid);
```

Comment:
```suggestion
            _popup.PopupEntity(Loc.GetString(Loc.GetString("alien-action-fail-plasma")), uid, uid);
```
Попап клиент на сервере не работает